compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -498,6 +498,21 @@ extern "C" LLVMTargetMachineRef LLVMRustCreateTargetMachine(
   Options.EmitStackSizeSection = EmitStackSizeSection;
 
   if (ArgsCstrBuff != nullptr) {
+#if LLVM_VERSION_GE(20, 0)
+    int buffer_offset = 0;
+    assert(ArgsCstrBuff[ArgsCstrBuffLen - 1] == '\0');
+    std::string Arg0(ArgsCstrBuff);
+    buffer_offset = Arg0.size() + 1;
+    std::string ArgsCppStr(ArgsCstrBuff + buffer_offset, ArgsCstrBuffLen - 1);
```

Comment:
So, this PR looks overall fine but this style of constructor is recommended against by several C++ style guides, including e.g. [Abseil's](https://abseil.io/tips/88), which is why I removed all instances of it from rustc_llvm in favor of assignment constructors, which look like... well, more like all the rest of our code in this repo: https://github.com/rust-lang/rust/pull/122062

Reply:
Fair enough - I never can remember the C++ right choices. Fixed.